### ISSUE TITLE
[Feature Request] The option to pull up categories to fill vertical gaps

### ISSUE THREAD

**Issue by Brancliff** [pawelmalak/flame#269]
Not sure how to best describe this one in words, so here's a visual expression with a 'before' on the left and an 'after' on the right

![pull up](https://user-images.githubusercontent.com/22332774/147332920-5efcbf74-bfaf-4b5d-bad9-d689b7533abd.png)

If you have one category with a lot more bookmarks than the other, it can create a lot of empty space to scroll through. This space could be filled up by lifting categories that are further down the page and placing them in that empty space.

**Comment by katacarbix** [comment]
I agree that this would be nice, but layouts like that can be rather resource-intensive when done in Javascript and there's no standard way to do it in CSS either.

[Recent versions of Firefox have an experimental solution](https://caniuse.com/?search=grid-template-rows%3A%20masonry) that is available by setting `layout.css.grid-template-masonry-value.enabled` to `true` in `about:config`. Then add the following to Flame's custom CSS:

```css
[class^=BookmarkGrid_BookmarkGrid__] {
  grid-template-rows: masonry;
}
```
